### ISSUE TITLE
Adding description to fields.Nested makes it display as String field

### ISSUE THREAD

**Issue by AFlowOfCode** [python-restx/flask-restx#479]
This occurs in the Swagger visualization under the endpoint. In the Models section it is always correctly displayed as an object, but note that when there is a description it seems to lose the model name as well. When there is no description it correctly displays `NestedObj` next to the arrow which expands the nested model.

Since [Nested inherits from Raw](https://flask-restx.readthedocs.io/en/latest/api.html#flask_restx.fields.Nested), I would expect `description` to be a valid keyword argument & be initialized in this line: `super(Nested, self).__init__(**kwargs)`. This seems to be a bug, or if I'm doing something wrong, at minimum it is confusing and unexpected behavior.

```
nested_obj = api.model('NestedObj', {
    'nested_field': fields.String()    
})

parent_obj = api.model('ParentObj', {
    'parent_field': fields.String(),
    'nested_obj': fields.Nested(nested_obj, required=True, description="A nested object")
})
```
![nested description converts display to string](https://user-images.githubusercontent.com/14116237/194406122-ebfda21a-fa37-4fa1-adc6-a5636cc1c617.png)
![nested description converts display to string 2](https://user-images.githubusercontent.com/14116237/194406867-ec5758ce-3acc-4ca0-b195-d4ad124b32b0.png)

```
nested_obj = api.model('NestedObj', {
    'nested_field': fields.String()    
})

parent_obj = api.model('ParentObj', {
    'parent_field': fields.String(),
    'nested_obj': fields.Nested(nested_obj, required=True)
})
```
![nested description converts display to string (without description)](https://user-images.githubusercontent.com/14116237/194406455-89be5efc-b16d-44f6-88a4-285ad3e85059.png)
![nested description converts display to string (without description) 2](https://user-images.githubusercontent.com/14116237/194406626-d0eba7bd-0200-4f71-8d2a-809cf8a03713.png)

**Comment by peter-doggart** [comment]
Which method are you using to document the models on your endpoints? 

When I take your first batch of code into a sample app, I get the expected result.

```python
from flask import Flask
from flask_restx import Api, Resource, fields

app = Flask(__name__)
api = Api(app, version='1.0', title='Test API',
    description='A simple Test API',
)

ns = api.namespace('test', description='Test operations')

nested_obj = api.model('NestedObj', {
    'nested_field': fields.String()    
})

parent_obj = api.model('ParentObj', {
    'parent_field': fields.String(),
    'nested_obj': fields.Nested(nested_obj, required=True, description="A nested object")
})


@ns.route('/')
class Test(Resource):
    '''Test'''
    @ns.expect(parent_obj)
    def post(self):
        '''Create a new task'''
        return parent_obj, 201


if __name__ == '__main__':
    app.run(debug=True)
```
![Screenshot from 2022-10-07 10-52-43](https://user-images.githubusercontent.com/10991724/194526903-ef858402-5095-41bc-a74e-d7233b4f2c2c.png)

**Comment by AFlowOfCode** [comment]
I had it like this:
```
@ns.route('/')
@ns.expect(parent_obj)
class Test(Resource):
    '''Test'''
    def post(self):
        '''Create a new task'''
        return parent_obj, 201
```

But thanks to your example I discovered when I put the decorator directly above the endpoint definition vs the class it does work as expected. 

Is it specifically incorrect to put it above the class? Generally speaking putting the decorator above the class always works, except in this instance apparently.

**Comment by peter-doggart** [comment]
> Is it specifically incorrect to put it above the class?

@AFlowOfCode No, as far as I'm aware, both should work the same, so this is definitely a bug! Thanks for providing how to reproduce, I will try and find some time to take a look / fix. It definitely has to be a small difference in how the `swagger.json` file is being populated somewhere!

**Comment by AFlowOfCode** [comment]
Thanks & thanks for demonstrating the workaround in the meantime.
